### PR TITLE
Resolves #35, Resolves #34, Resolves #33, Resolves #32, Resolves #31, Resolves #30, Resolves #29

### LINKED ISSUE #29: Merchant Items Index: 5 most popular items
As a merchant
When I visit my items index page
Then I see the names of the top 5 most popular items ranked by total revenue generated
And I see that each item name links to my merchant item show page for that item
And I see the total revenue generated next to each item name

Notes on Revenue Calculation:
- Only invoices with at least one successful transaction should count towards revenue
- Revenue for an invoice should be calculated as the sum of the revenue of all invoice items
- Revenue for an invoice item should be calculated as the invoice item unit price multiplied by the quantity (do not use the item unit price)

### LINKED ISSUE #30: Merchant Item Create
As a merchant
When I visit my items index page
I see a link to create a new item.
When I click on the link,
I am taken to a form that allows me to add item information.
When I fill out the form I click ‘Submit’
Then I am taken back to the items index page
And I see the item I just created displayed in the list of items.
And I see my item was created with a default status of disabled.

### LINKED ISSUE #31: Merchant Items Grouped by Status
As a merchant,
When I visit my merchant items index page
Then I see two sections, one for "Enabled Items" and one for "Disabled Items"
And I see that each Item is listed in the appropriate section

### LINKED ISSUE #32: Merchant Item Disable/Enable
As a merchant
When I visit my items index page
Next to each item name I see a button to disable or enable that item.
When I click this button
Then I am redirected back to the items index
And I see that the items status has changed

### LINKED ISSUE #33: Merchant Item Update
As a merchant,
When I visit the merchant show page of an item
I see a link to update the item information.
When I click the link
Then I am taken to a page to edit this item
And I see a form filled in with the existing item attribute information
When I update the information in the form and I click ‘submit’
Then I am redirected back to the item show page where I see the updated information
And I see a flash message stating that the information has been successfully updated.

### LINKED ISSUE #34: Merchant Items Show Page
As a merchant,
When I click on the name of an item from the merchant items index page,
Then I am taken to that merchant's item's show page (/merchant/merchant_id/items/item_id)
And I see all of the item's attributes including:

- Name
- Description
- Current Selling Price

### LINKED ISSUE #35: Merchant Items Index Page
As a merchant,
When I visit my merchant items index page ("merchant/merchant_id/items")
I see a list of the names of all of my items
And I do not see items for any other merchant

### PR DESCRIPTION
**Pull request by arnaldoaparicio**
This update is a refactor for items controller to make it adhere to restfulness as well as changing the way the Enable/Disable button behaves.

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -1,82 +1,47 @@
 class ItemsController < ApplicationController
   # before_action :set_item, only: %i[ show edit update destroy ]
 
-  # GET /items or /items.json
```
Comment (tjroeder) [inline app/controllers/items_controller.rb:4]: These comments are a little excessive by default.

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -1,82 +1,47 @@
...
     @merchant = Merchant.find(params[:merchant_id])
   end
 
-  # POST /items or /items.json
   def create
-    merchant = Merchant.find(params[:merchant_id])
     item = Item.new(item_params)
     item.save
-    redirect_to "/merchants/#{merchant.id}/items"
+    redirect_to merchant_items_path(params[:merchant_id])
```
Comment (tjroeder) [inline app/controllers/items_controller.rb:25]: Nice change this is a lot cleaner.

File: app/controllers/items_controller.rb [hunk 3]
```diff
@@ -1,82 +1,47 @@
...
   end
 
-  # PATCH/PUT /items/1 or /items/1.json
   def update
-    merchant = Merchant.find(params[:merchant_id])
     item = Item.find(params[:id])
-    if params[:name] == "" || params[:description] == "" || params[:unit_price] == ""
-      redirect_to "/merchants/#{merchant.id}/items/#{item.id}/edit"
-      flash[:alert] = "You cannot leave any field blank."
+    if params[:status] == 'Disabled'
```
Comment (tjroeder) [inline app/controllers/items_controller.rb:30]: Nice refactor to make the #status_change more RESTful by adding to the #update action.

**Review by WadeNaughton — APPROVED** [review]

File: app/views/items/index.html.erb [hunk 4]
```diff
@@ -8,14 +8,17 @@
 <h2>Enabled Items</h2>
 <% @merchant.enabled_items.each do |item| %>
     <p><%= link_to item.name, merchant_item_path(@merchant, item) %>
-      <%= button_to "Enable/Disable", "/merchants/#{@merchant.id}/items/#{item.id}/status_change", method: :patch %></p>
+    <% if item.status == "Enabled" %>
+      <%= button_to "Disable",  "/merchants/#{@merchant.id}/items/#{item.id}", method: :patch, params: {status: 'Enabled'} %></p>
```
Comment (tjroeder) [inline app/views/items/index.html.erb:12]: The hidden param is a great addition for the #update action.

File: spec/features/items/edit.html.erb_spec.rb [hunk 5]
```diff
@@ -4,7 +4,7 @@
   before(:each) do
     @merchant_1 = Merchant.create!(name: 'Joe Shmoe')
     @item_1 = @merchant_1.items.create!(name: 'Light Machine', description: 'Revolutionary Device', unit_price: 999)
-    visit "/merchants/#{@merchant_1.id}/items/#{@item_1.id}/edit"
+    visit edit_merchant_item_path(@merchant_1, @item_1)
```
Comment (tjroeder) [inline spec/features/items/edit.html.erb_spec.rb:7]: This is a lot easier to read.

**Review by tjroeder — APPROVED** [review]
Changes look really good for the refactor and the removal of the #status_change action.

**Pull request merged**